.github/workflows/step_tests-pip.yml

```diff
@@ -1,54 +1,69 @@
 name: test-pip
-run-name: Run tests using Pip
+run-name: Run main tests using Pip
 
 on:
   workflow_call:
+    inputs:
+      coverage:
+        type: boolean
+        description: Upload coverage to CodeCov
+        default: true
 
-concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}-pip-test
-  cancel-in-progress: true
+permissions:
+  contents: read
 
 jobs:
   test-pip:
+    name: >
+      🐍 ${{ matrix.python-version }}
+      ${{ matrix.dependency_type && format('({0})', matrix.dependency_type) }}
+      ${{ matrix.markdown-it-py && format('(markdown-it-py {0})', matrix.markdown-it-py) }}
+      ${{ matrix.quarto && '(Quarto)' }}
+      ${{ matrix.no_kernel && '(No kernel)' }}
     runs-on: ubuntu-latest
 
     strategy:
+      fail-fast: false
       matrix:
         python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12"]
-        markdown-it-py-version: ["~=3.0"]
-        pip-flags: [""]
         include:
+          # Test pre-release versions
           - python-version: "3.x"
-            pip-flags: "--pre --upgrade --upgrade-strategy eager"
+            dependency_type: "pre"
+          # Test minimum markdown-it-py supported (otherwise the most recent version is used)
           - python-version: "3.x"
-            markdown-it-py-version: "~=2.0"
+            markdown-it-py: "~=2.0"
           - python-version: "3.x"
-            markdown-it-py-version: ""
+            markdown-it-py: false
```

Comment:
Testing that Jupytext works without `markdown-it-py` in the CI seems very hard to achieve! It does not seem to be working on `main`, but I am not sure it is working here neither. I will have a look.

Cf. https://github.com/mwouts/jupytext/actions/runs/7506308274/job/20437528721 which does not run the 'uninstall markdown-it-py' step:
![image](https://github.com/mwouts/jupytext/assets/29915202/c1d5b558-ebdb-496b-a164-07f664e2dc3c)